Updated User.md and Networking.md with needed changes.

Jump-Suit opened the pull request:
Added OnCurrentUserUpdate example and updated Networking.md with needed changes.

Jump-Suit committed:
Deprecated Match Secret and Instance Bool Accordingly

Commit:
Merge pull request #11 from discordapp/master

move inviter field to invite object (#1270)

Jump-Suit committed:
Needed changes to User and Networking

Updated Users.md with OnCurrentUserUpdate Example and Networking.md with needed changes!

Jump-Suit committed:
Merge branch 'master' of https://github.com/Jump-Suit/discord-api-docs

msciotti commented on docs/game_sdk/Users.md:
```diff
@@ -142,6 +142,21 @@ Fires when the `User` struct of the currently connected user changes. They may h
+Console.WriteLine(currentUser.Username);
+Console.WriteLine(currentUser.Id);
+Console.WriteLine(currentUser.Discriminator);
+Console.WriteLine(currentUser.Avatar);
```
Some spacing issues here; this should be formatted like C# (4 spaces I think?)

Jump-Suit replied:
I did notice that as well, but was unsure whether to keep it as such or not... Since I just grabbed it from Program.cs .

Jump-Suit replied:
Spaced out! Respectively.

msciotti commented on docs/game_sdk/Networking.md:
```diff
@@ -316,16 +316,18 @@ lobbyManager.OnMemberUpdate += (lobbyId, userId) =>
 
-// Connect to lobby with an id of 12345 and a secret of "password"
+// Connect to lobby with an id of 12345 and a secret of "password" 
+// This is used when Lobby Searching and a User wants to join remotely through this!
```
Wording:

> "This may occur in a lobby search, when a user needs to input a password to join a lobby"

Jump-Suit replied:
Will Change it to that! I just wanted to give more insight in that part of the function, since that whole example imo is just snippets mashed together...?

msciotti commented on docs/game_sdk/Networking.md:
```diff
@@ -285,20 +285,20 @@ var networkManager = discord.GetNetworkManager();
-  txn.SetMetadata("route", route);
-  lobbyManager.UpdateMember(lobbyId, me.Id, txn, (result =>
+  var txn = lobbyManager.GetMemberUpdateTransaction(lobbyId, currentUser.Id);
+  txn.SetMetadata("metadata.route", route);
```
You don't need to explicitly have `metadata.` when setting metadata, only when `GET`ing it.

Jump-Suit replied:
From my experience using the metadata, I did previously set it as just ` txn.SetMetadata("route", route);` or for the peer_Id as well, but it would result from discord saying 'Not Found'. So i set it as that to completely make sure the wording is the same, and in my case/experience Discord returns the peer_id and route properly... So hence why I have it as such.

msciotti replied:
I’d like to test this myself before changing the docs just to be sure if that’s alright. Thanks for your patience! I’ll test this tomorrow.

Jump-Suit replied:
Anything new regarding this?

msciotti replied:
@Amorelandra was testing for a repro. Any luck?

night replied:
When accessing `metadata` methods you should not need a `metadata.` prefix. Metadata is just a key-value store, so if you set `metadata.route` then you would have to retrieve with `metadata.route` in `GetMemberMetadataValue`.

The `metadata.` prefix comes into play when you are doing filtering/sorting on lobbies, since `metadata.` triggers a comparison against the metadata on a lobby

Jump-Suit replied:
> 
> 
> When accessing `metadata` methods you should not need a `metadata.` prefix. Metadata is just a key-value store, so if you set `metadata.route` then you would have to retrieve with `metadata.route` in `GetMemberMetadataValue`.
> 
> The `metadata.` prefix comes into play when you are doing filtering/sorting on lobbies, since `metadata.` triggers a comparison against the metadata on a lobby

I see Night! That DOES make more sense how the prefix comes into play, but what i still don't get is why without that prefix being set to the data of the user, it gives 'result: not found' this is the workaround I've come across... So not sure...? I'd be glad to reverse this if that is the case.

msciotti replied:
I can confirm that what night said is true. Here is the example code for a Unity sample app:

```cs
discord = new Discord.Discord(APP_ID, (System.UInt64)Discord.CreateFlags.Default);
    discord.SetLogHook(Discord.LogLevel.Debug, (Discord.LogLevel level, string message) =>
    {
      Debug.Log(message);
    });

    lobbyManager = discord.GetLobbyManager();
    var txn = lobbyManager.GetLobbyCreateTransaction();
    txn.SetMetadata("mmr", "1000");

    lobbyManager.CreateLobby(txn, (Discord.Result res, ref Discord.Lobby lobby) =>
    {
      Debug.Log(lobby.Id);
      Debug.Log("Metadata: " + lobbyManager.GetLobbyMetadataValue(lobby.Id, "mmr"));
    });
```

![image](https://user-images.githubusercontent.com/4193854/72394861-8765d100-36ec-11ea-980a-61b03f335425.png)

so yea please update accordingly :)

Jump-Suit replied:
Reversing!

msciotti reviewed (changes requested):
Left some comments to address

Jump-Suit committed:
Revised Changes to GameSDK Docs

Spaced out certain lines in User.md and Updated Networking.md with requested comment change. Kept 'metadata.' As it DOES make it function properly in my testing compared to having it not there.

Jump-Suit committed:
Reversed SetMetadata via request.

Reversed SetMetadata so that 'metadata.' isn't included.

Commit:
Merge pull request #12 from discordapp/master

Merge latest Commits to Master.

msciotti commented on docs/game_sdk/Networking.md:
```diff
@@ -338,10 +340,10 @@ lobbyManager.ConnectLobby(12345, "password", (Discord.Result x, ref Discord.Lobb
   // Metadata is stored as a string, so we need to make it an integer for OpenChannel
   otherUserPeerId = System.Convert.ToUInt64(rawPeerId);
-  var otherRoute = lobbyManager.GetMemberMetadataValue(lobby.Id, member.Id, "route");
+  var otherRoute = lobbyManager.GetMemberMetadataValue(lobby.Id, member.Id, "metadata.route");
```
Can we drop `metadata.` here as well.

Jump-Suit replied:
Done!

Jump-Suit committed:
Removed the GET 'metadata' ones as well

GET wasn't nessecary either! added (UInt64) to a create parameter that is needed.

Jump-Suit committed:
Merge branch 'master' of https://github.com/Jump-Suit/discord-api-docs
